pkg/chains/formats/intotoite6/taskrun/material.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package taskrun
+
+import (
+	"reflect"
+	"strings"
+
+	slsa "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.2"
+	"github.com/tektoncd/chains/pkg/artifacts"
+	"github.com/tektoncd/chains/pkg/chains/formats/intotoite6/attest"
+	"github.com/tektoncd/chains/pkg/chains/objects"
+	"github.com/tektoncd/pipeline/pkg/apis/resource/v1alpha1"
+	"go.uber.org/zap"
+)
+
+const (
+	uriSeparator    = "@"
+	digestSeparator = ":"
+)
+
+func isDuplicateMaterial(mats []slsa.ProvenanceMaterial, m slsa.ProvenanceMaterial) bool {
+	for _, mat := range mats {
+		if reflect.DeepEqual(m, mat) {
```

Comment:
Probably worth documenting (and maybe even worth adding a test for) since I had to go double check on this behavior - this should be a stable check because the implementation of DeepEqual is agnostic to key ordering.

Reply:
I'm also wondering if we can turn this into more like a set so we don't need to iterate through the list of materials each time 🤔 

I don't think we can use a map key of maps, so you might need to do something like json serialize it first and use a `map[string]bool`